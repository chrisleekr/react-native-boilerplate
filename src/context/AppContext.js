import React, {createContext} from 'react';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';

import {ThemeProvider} from '../theme/useTheme';
import store, {persistor} from '../store/store';

import Loading from '../component/general/loading';

const AppContext = createContext();

const AppContextProvider = ({children}) => {
  const appInitialState = {};

  return (
    <AppContext.Provider value={appInitialState}>
      <ThemeProvider>
        <Provider store={store}>
          <PersistGate loading={<Loading />} persistor={persistor}>
            {children}
          </PersistGate>
        </Provider>
      </ThemeProvider>
    </AppContext.Provider>
  );
};

export default AppContextProvider;
