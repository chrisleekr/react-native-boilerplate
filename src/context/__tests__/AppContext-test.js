import React from 'react';
import {Text} from 'react-native';

import {cleanup, render} from '@testing-library/react-native';

import AppContext from '../AppContext';

let rendered;

describe('AppContext', () => {
  afterEach(cleanup);

  beforeEach(() => {
    rendered = render(
      <AppContext>
        <Text testID="text">Test message</Text>
      </AppContext>
    );
  });

  it('matches snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should not show loading screen', () => {
    expect(rendered.queryByTestId('loading')).toBeFalsy();
  });

  it('shows text children', () => {
    expect(rendered.queryByTestId('text')).toBeTruthy();
  });
});
