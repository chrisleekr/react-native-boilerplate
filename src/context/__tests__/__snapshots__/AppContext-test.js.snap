// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContext matches snapshot 1`] = `
<Text
  testID="text"
>
  Test message
</Text>
`;
